phenopacketlab-io/src/main/java/org/monarchinitiative/phenopacketlab/io/VariantValidatorFunctionalAnnotationService.java

```diff
@@ -0,0 +1,137 @@
+            case ALL -> encodeHgvsFromVariantValidator(build, description, "all").stream();
+        };
+    }
+
+    /**
+     * Query the Variant validator <a href="https://rest.variantvalidator.org/">API</a> to build a Variant object
+     * Method similar to
+     * <a href="https://github.com/monarch-initiative/pyphetools/blob/main/pyphetools/creation/variant_validator.py">Pyphetools</a>
+     *
+     * @param build genome build
+     * @param hgvs hgvs description
+     * @param transcript transcript can be
+     * @return list of VariantMetadata objects
+     */
+    private List<VariantMetadata> encodeHgvsFromVariantValidator(String build, String hgvs, String transcript) {
+        List<VariantMetadata> results = new ArrayList<>();
+        try {
+            String variantValidatorUrl = "https://rest.variantvalidator.org/VariantValidator/variantvalidator/" + build + "/" + URLEncoder.encode(hgvs, StandardCharsets.UTF_8) + "/" + URLEncoder.encode(transcript, StandardCharsets.UTF_8) + "?content-type=application/json";
+            // create a request
+            HttpRequest request = HttpRequest.newBuilder()
+                    .version(HttpClient.Version.HTTP_2)
+                    .uri(URI.create(variantValidatorUrl))
+                    .GET()
+                    .header("accept", "application/json")
+                    .build();
+
+            // use the client to send the request
+            CompletableFuture<HttpResponse<String>> response = client.sendAsync(request, HttpResponse.BodyHandlers.ofString());
+            // This blocks until the request is complete
+
+            String result1 = response.thenApply(HttpResponse::body).get(5, TimeUnit.SECONDS);
+
+            JsonNode tree = objectMapper.readTree(result1);
+            // Put map into variant object
+            Iterator<String> names = tree.fieldNames();
+
+            while (names.hasNext()) {
+                String key = names.next();
+                if (!key.equals("flag") && !key.equals("metadata")) {
+                    VariantMetadata variant = new VariantMetadata(build);
+
+                    JsonNode node = tree.get(key);
+                    JsonNode geneNode = node.get("gene_ids");
+                    if (geneNode != null) {
+                        String hgncId = geneNode.get("hgnc_id").asText();
+                        variant.setHgncId(hgncId);
+                    }
+
+                    String geneSymbol = node.get("gene_symbol").asText();
+                    variant.setGeneSymbol(geneSymbol);
+
+                    JsonNode primaryLociNode = node.get("primary_assembly_loci");
+                    if (primaryLociNode != null) {
+                        if (!primaryLociNode.has(build) && !primaryLociNode.has(build.toLowerCase())) {
+                            throw new Exception("Could not identify" + build + " in Variant Validator response.");
```

Comment:
I am not sure this is the way how we should handle the exceptions. It sounds like an error to not get a response with the genome build we asked for, and I'm not sure the user can do anything about it. I think we should log an error, `continue`. Depending on whether we get at least one good item, we return something or an empty list. The empty list can indicate that something went wrong and that the frontend should display some generic message ("Sorry, something went wrong"). Alternatively, we should can add an exception to the `FunctionalVariantAnnotationService` API and handle it at the caller side. 

Please note that I changed the line 131.. such that the exception is logged and swallowed. The exception handling we're discussing here would need to include that case too.

@belkassaby what are your thoughts?